internal/factsengine/gatherers/saphostctrl.go

```diff
@@ -0,0 +1,142 @@
+package gatherers
+
+import (
+	"regexp"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/trento-project/agent/pkg/factsengine/entities"
+	"github.com/trento-project/agent/pkg/utils"
+)
+
+const (
+	SapHostCtrlGathererName               = "saphostctrl"
+	saphostCtrlListInstancesParsingRegexp = `^\s+Inst Info\s*:\s*([^-]+?)\s*-\s*(\d+)\s*-\s*([^,]+?)\s*-\s*(\d+),\s*patch\s*(\d+),\s*changelist\s*(\d+)$`
+	saphostCtrlPingParsingRegexp          = `(SUCCESS) \( *(\d+) usec\)`
+)
+
+// nolint:gochecknoglobals
+var (
+	SapHostCtrlCommandError = entities.FactGatheringError{
+		Type:    "saphostctrl-cmd-error",
+		Message: "error executing saphostctrl command",
+	}
+	SapHostCtrlUnsupportedFunction = entities.FactGatheringError{
+		Type:    "saphostctrl-func-error",
+		Message: "requested function not whitelisted",
+	}
+	SapHostCtrlParseError = entities.FactGatheringError{
+		Type:    "saphostctrl-parse-error",
+		Message: "error while parsing saphostctrl output",
+	}
+)
+
+type SapHostCtrlGatherer struct {
+	executor utils.CommandExecutor
+}
+
+func NewDefaultSapHostCtrlGatherer() *SapHostCtrlGatherer {
+	return NewSapHostCtrlGatherer(utils.Executor{})
+}
+
+func NewSapHostCtrlGatherer(executor utils.CommandExecutor) *SapHostCtrlGatherer {
+	return &SapHostCtrlGatherer{
+		executor: executor,
+	}
+}
+
+func (g *SapHostCtrlGatherer) Gather(factsRequests []entities.FactRequest) ([]entities.Fact, error) {
+	facts := []entities.Fact{}
+	log.Infof("Starting saphostctrl facts gathering process")
+
+	for _, factReq := range factsRequests {
+		var fact entities.Fact
+
+		if !whitelisted()[factReq.Argument] {
```

Comment:
We don't really need to run the `whitelisted` function all the times in the loop.
We could assign it to a variable before going in the loop